docs/dev/query/msq/execution.md

```diff
@@ -0,0 +1,48 @@
+# Multi stage query execution
```

Comment:
I think it would be good to:
- decide on a single term and acronyms, e.g.  multi stage query engine/MSQE vs multi stage query/MSQ, single stage query/SSQ vs single stage query engine/SSQE 
- define them somewhere 
At the moment they're used assuming reader knows what they mean.